website/static/js/osfHelpers.js

```diff
@@ -357,6 +358,57 @@ ko.bindingHandlers.anchorScroll = {
 };
 
 /**
+ * Adds class returned from iconmap to the element. The value accessor should be the 
+ * category of the node.
+ * Example:
+ * <span data-bind="getTheIcon: 'analysis'"></span>
+ */
+ko.bindingHandlers.getTheIcon = {
+    init: function(elem, valueAccessor) {
+        var icon;
+        var category = valueAccessor();
+        if (Object.keys(iconmap.componentIcons).indexOf(category) >=0 ){
+            icon = iconmap.componentIcons[category];
+        }
+        else {
+            icon = iconmap.projectIcons[category];
+        }
+        $(elem).addClass(icon);
+    },
+    update: function(elem, valueAccessor) {
```

Comment:
It would be if the category were to change while you were viewing the project. If you were to change it yourself then you would have to leave the page anyway but if you were viewing a project and a collaborator changed the component type than it would need to update. Am I correct in this assumption @samanehsan?